rust/lancedb/src/table.rs

```diff
@@ -1974,6 +1992,12 @@ impl BaseTable for NativeTable {
                 message: "Multi-column (composite) indices are not yet supported".to_string(),
             });
         }
+        if opts.wait_timeout.is_some() {
+            return Err(Error::InvalidInput {
+                message: "wait_timeout must be None for NativeTable".to_string(),
+            });
+        }
```

Comment:
I think it would be useful to allow a `wait_timeout` on `NativeTable` and just return immediately.  Partly because users can still create their own background indexing processes.  But mostly because this will make it easier for users to design code that can switch between cloud / native seamlessly.

Reply:
yeah that makes sense. I was trying to guard against the case where a user sets the timeout but it does nothing, but I guess we can't really avoid that unless the local/remote APIs diverge.